pyproject.toml

```diff
@@ -246,10 +246,11 @@ docstring-code-format = true
 [tool.pytest.ini_options]
 addopts = "--durations=30 --quiet -r xXs --color=yes --import-mode=importlib"
 filterwarnings = [
-    # NOTE: the last matching option is used
-    "ignore::Warning",               # Ignore all Warning
-    "default::FutureWarning",        # Show FutureWarnings
-    "default::DeprecationWarning",   # Show DeprecationWarnings
+    # NOTE: the LAST matching option would be used
+    "ignore::UserWarning",                                              # Ignore UserWarning
+    "error:We strongly encourage explicit `encoding`:EncodingWarning",  # Mark `zopen` EncodingWarning as error
+    # TODO: remove the following filter once `monty.io` dropped custom warning
+    "error:We strongly encourage explicit `encoding`:monty.io.EncodingWarning",
```

Comment:
I believe this is the correct filter now:
```
=============================================================================== FAILURES ================================================================================
____________________________________________________________________ TestACstrExtractor.test_extract ____________________________________________________________________

self = <tests.io.pwmat.test_inputs.TestACstrExtractor testMethod=test_extract>

    def test_extract(self):
        filepath = f"{TEST_DIR}/atom.config"
        ac_extractor = ACExtractor(file_path=filepath)
>       with zopen(filepath, mode="rt") as file:

/Users/yang/developer/pymatgen/tests/io/pwmat/test_inputs.py:50: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

filename = '/Users/yang/developer/pymatgen/src/../tests/files/io/pwmat/atom.config', mode = 'rt', kwargs = {}, _deadline = '2025-06-01'

    def zopen(
        filename: Union[str, Path],
        /,
        mode: str | None = None,
        **kwargs: Any,
    ) -> IO | bz2.BZ2File | gzip.GzipFile | lzma.LZMAFile:
        """
        This function wraps around `[bz2/gzip/lzma].open` and `open`
        to deal intelligently with compressed or uncompressed files.
        Supports context manager:
            `with zopen(filename, mode="rt", ...)`.
    
        Important Notes:
            - Default `mode` should not be used, and would not be allow
                in future versions.
            - Always explicitly specify binary/text in `mode`, i.e.
                always pass `t` or `b` in `mode`, implicit binary/text
                mode would not be allow in future versions.
            - Always provide an explicit `encoding` in text mode.
    
        Args:
            filename (str | Path): The file to open.
            mode (str): The mode in which the file is opened, you MUST
                explicitly specify "b" for binary or "t" for text.
            **kwargs: Additional keyword arguments to pass to `open`.
    
        Returns:
            TextIO | BinaryIO | bz2.BZ2File | gzip.GzipFile | lzma.LZMAFile
        """
        # Deadline for dropping implicit `mode` support
        _deadline: str = "2025-06-01"
    
        # Warn against default `mode`
        # TODO: remove default value of `mode` to force user to give one after deadline
        if mode is None:
            warnings.warn(
                "We strongly discourage using a default `mode`, it would be"
                f"set to `r` now but would not be allowed after {_deadline}",
                FutureWarning,
                stacklevel=2,
            )
            mode = "r"
    
        # Warn against implicit text/binary `mode`
        # TODO: replace warning with exception after deadline
        elif not ("b" in mode or "t" in mode):
            warnings.warn(
                "We strongly discourage using implicit binary/text `mode`, "
                f"and this would not be allowed after {_deadline}. "
                "I.e. you should pass t/b in `mode`.",
                FutureWarning,
                stacklevel=2,
            )
    
        # Warn against default `encoding` in text mode
        if "t" in mode and kwargs.get("encoding", None) is None:
>           warnings.warn(
                "We strongly encourage explicit `encoding`, "
                "and we would use UTF-8 by default as per PEP 686",
                category=EncodingWarning,
                stacklevel=2,
            )
E           EncodingWarning: We strongly encourage explicit `encoding`, and we would use UTF-8 by default as per PEP 686

/Users/yang/developer/pymatgen/venv312/lib/python3.12/site-packages/monty/io.py:81: EncodingWarning
=========================================================================== warnings summary ============================================================================
venv312/lib/python3.12/site-packages/numpy/testing/_private/utils.py:1247
  /Users/yang/developer/pymatgen/venv312/lib/python3.12/site-packages/numpy/testing/_private/utils.py:1247: EncodingWarning: 'encoding' argument not specified.
    output = subprocess.run(cmd, capture_output=True, text=True)

tests/io/pwmat/test_inputs.py::TestACstrExtractor::test_extract
  /Users/yang/developer/pymatgen/src/pymatgen/io/pwmat/inputs.py:139: EncodingWarning: 'encoding' argument not specified
    with open(self.atom_config_path) as file:

tests/io/pwmat/test_inputs.py::TestACstrExtractor::test_extract
  /Users/yang/developer/pymatgen/src/pymatgen/io/pwmat/inputs.py:154: EncodingWarning: 'encoding' argument not specified
    with open(self.atom_config_path) as file:

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================================================================= slowest 30 durations ==========================================================================

(3 durations < 0.005s hidden.  Use -vv to show these durations.)
1 failed, 3 warnings in 0.27s
```

Reply:
I have tried to find this out from the code and reading things up and therefore I would say yes, it will catch EncodingWarnings from monty.io.zopen.